client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/generated-design-picker-web-preview.tsx

```diff
@@ -45,6 +51,7 @@ const GeneratedDesignPickerWebPreview: React.FC< GeneratedDesignPickerWebPreview
 			autoHeight
 			siteId={ site?.ID }
 			url={ site?.URL }
+			fixedViewportWidth={ ! isMobile ? DEFAULT_VIEWPORT_WIDTH : MOBILE_VIEWPORT_WIDTH }
```

Comment:
Do we need to set the value on mobile?

Reply:
`fixedViewportWidth` needs a value so that the `resize` event listener is bound. See: https://github.com/Automattic/wp-calypso/pull/64316/files#diff-7f6337b08cbe32374f946901312eda1ba5a282f898f442dd258ba2275c5341fdR44 

Do you have any idea as to how to improve this?

Reply:
I think we have to detect the `fixedViewportWidth` changes from/to `undefined`
* If the value changes from `undefined`, we need to bind the resize listener and trigger it immediately
* if the value changes to `undefined`, we need to unbind the resize listener and reset the `iframeScaleRatio` and the `iframeStyle`.

Does it make sense?